### PR TITLE
devilutionx: minor cleanup after review

### PR DESCRIPTION
**Pull request by aanderse**
###### Motivation for this change
thank you very much @SuperSandro2000 for your suggestions in https://github.com/NixOS/nixpkgs/pull/144569!

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

ping @karolchmist

### CODE REVIEW

**Review by thiagokokada — APPROVED** [review]
Result of `nixpkgs-review pr 144999` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

No-op, good signal.

**Comment by thiagokokada** [comment]
Will leave the merge part to @SuperSandro2000 just in case he remembers something else.

File: pkgs/applications/misc/smpq/default.nix [hunk 1]
```diff
@@ -1,11 +1,11 @@
 { lib, stdenv, fetchurl, cmake, StormLib }:
 
-stdenv.mkDerivation {
+stdenv.mkDerivation rec {
   pname = "smpq";
   version = "1.6";
 
   src = fetchurl {
-    url = "https://launchpad.net/smpq/trunk/1.6/+download/smpq_1.6.orig.tar.gz";
+    url = "https://launchpad.net/smpq/trunk/${version}/+download/${pname}_${version}.orig.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/smpq/default.nix:8]: Not a big fan of substituting pname but not that important.

**Comment by karolchmist** [comment]
Am I the only one that can't run it with this PR?

I get :

```
Couldn't find matching render driver

The error occured at: SourceX/display.cpp line 218
```

It works if I build the [previous PR](https://github.com/NixOS/nixpkgs/pull/144569)

Here is the screenshot:

2021-11-09-174439_1065x882_scrot
?

**Comment by aanderse** [comment]
Eh? As @thiagokokada mentioned:
> No-op, good signal.

This PR didn't actually change anything :eyes: You're sure the previous PR builds? I force pushed after your initial test...

**Comment by karolchmist** [comment]
I've just re-built both PRs with `nixpkgs-review pr XXX`, and https://github.com/NixOS/nixpkgs/pull/144569 works, whereas this one still fails. Maybe if there is a "cache" or something I should remove in order to make it build from scratch? 

If it's only me, then nevermind...

**Comment by SuperSandro2000** [comment]
If you do nix-collect-garbage the build cache should be removed.

Also you only need a GITHUB_TOKEN if you want to interact with GitHub or are rate limited.

**Comment by aanderse** [comment]
@karolchmist please confirm when you get things working!

**Comment by thiagokokada** [comment]
> I've just re-built both PRs with `nixpkgs-review pr XXX`, and #144569 works, whereas this one still fails. Maybe there is a "cache" or something I should remove in order to make it build from scratch?
> 
> If it's only me, then nevermind...

Can you check the `/nix/store` hash. Here on my computer, using `master` branch, I have the following for `x86_64-linux` (that hits the Hydra cache so it doesn't build locally unless forced): `/nix/store/3d6x5xkk9hnlg7zl6dxx8ak53k31597q-devilutionx-1.3.0`

I ran the binary and looks fine. It showed one screen asking for the required `diabdat.mpq` or `spawn.mpq` files.

**Comment by karolchmist** [comment]
I found the reason: I had installed devilutionx with `nix-env -i`, so that's why I could find its executable in this PR. I uninstalled it and I could no longer run it. Everything is correct, sorry for the confusion!

**Pull request merged**